### ISSUE TITLE
Media selector broken for folders

### ISSUE THREAD

**Issue by jbasko** [strapi/strapi#17211]
### Required System information

- Node.js version: 18.12.1
- NPM version: 9.2.0
- Strapi version: 4.11.4
- Database: SQLite
- Operating system: macOS Monterey
- Is your project Javascript or Typescript: Javascript

### Describe the bug

My media library (backed by S3-like provider) is organised in folders. When managing the media library directly, it works fine. When editing a collection type object and I try to select an image residing in a folder, I only see images in the root folder of the media library.

### Steps to reproduce the behavior

1. Create a folder in Media library.
2. Upload an asset to that folder (say JPG image).
3. In admin, open an editor for an object of the content type which has a media type field.
4. Try to select an existing media object from a sub-folder. You will only see the root folder assets.

### Expected behavior

I expected the assets to be filtered by folder which I have clicked.

There is definitely some filtering happening as I don't see the root folders when I select a folder. But there is no filtering on assets as I only see the root folder assets (just not folders).

Media Library:

<img width="779" alt="Screenshot 2023-07-06 at 11 16 34" src="https://github.com/strapi/strapi/assets/632910/081fad3f-73d9-46d0-866c-36e6111ba001">

When selecting an image for an existing object, initially I see the correct contents of the root folder of the media library:

<img width="882" alt="Screenshot 2023-07-06 at 11 16 02" src="https://github.com/strapi/strapi/assets/632910/cf1c935d-78a3-481f-8698-a0dc3c58e14a">

However, when I select a folder, it does not show the contents of the folders (`logos` should contain 8 assets, and `people` should contain 1, but not the one being showed).

<img width="883" alt="Screenshot 2023-07-06 at 11 16 19" src="https://github.com/strapi/strapi/assets/632910/8619d4e0-f0a9-47e4-b7b6-3846fef7fc04">
<img width="881" alt="Screenshot 2023-07-06 at 11 16 10" src="https://github.com/strapi/strapi/assets/632910/eadc474f-d34e-43a1-9b51-52bde393c8bd">

**Comment by jbasko** [comment]
Looks like it was touched in 4.11.4 release: https://github.com/strapi/strapi/pull/17154

**Comment by JBG-brad** [comment]
I'm experiencing the same issue.  One thing to note is that when you use the search functionality, you do see assets regardless of the folder structure.

My current workaround is just to search for `.webp` to be able to select images.

**Comment by yusuf-saydullayev** [comment]
need to add this issue in 4.11.5 release

**Comment by de-shine** [comment]
Yes, Same issue for me. It is introduced as part of the latest strapi version v4.11.4.
Downgrading the version is the only solution for now.
